### DISCUSSION TITLE
AttributeError: 'GlobalStorage' object has no attribute 'float'

### DISCUSSION THREAD

**Discussion by kees317** [pyg-team/pytorch_geometric#5980] [Q&A]
Hello, I am trying to run my GCN on image dataset. I have used dataloader and this is the following output of the data loader:

![image](https://user-images.githubusercontent.com/68838553/201781605-e9a90138-4398-4691-89b9-8e5958ae4bb5.png)

below is my code for model and training of image data:
![image](https://user-images.githubusercontent.com/68838553/201782055-7e66092a-0c13-44fc-ae9e-ef229a1e6a93.png)
![image](https://user-images.githubusercontent.com/68838553/201782256-05d344c6-97a6-473c-9d10-5dd77602c6a2.png)

but i was getting the following error:
![image](https://user-images.githubusercontent.com/68838553/201782353-1e6699a8-4fe8-4894-af34-7b03614be806.png)
![image](https://user-images.githubusercontent.com/68838553/201782395-e121ed0e-ff6b-4c66-9d4c-c49e3b41d4dc.png)

then i changed my output to:
  output = model(data.float())

but now i am getting the following error:
![image](https://user-images.githubusercontent.com/68838553/201782535-d6be9871-7a50-4291-afe0-d0791318fc79.png)
![image](https://user-images.githubusercontent.com/68838553/201782623-aeee085c-c499-4a33-a1bf-424f359e7f83.png)

can anyone please tell me how to solve this?

**Comment by EdisonLeeeee** [comment]
```python
data.x = data.x.float()
```

**Reply by kees317** [reply]
i gave it like this:
![image](https://user-images.githubusercontent.com/68838553/202182940-4ca65829-5859-4b0f-8700-07f75f6e11f9.png)
but now i am getting this error:
![image](https://user-images.githubusercontent.com/68838553/202183051-c344245a-8a74-452c-937e-ffd85239c4a0.png)

**Reply by EdisonLeeeee** [reply]
Your `target` is a list of tensors instead of a tensor. Is that an intended behavior?
